data/src/main/java/hera/store/unit/TokenAccessUnit.java

```diff
@@ -11,8 +14,22 @@
 	public TokenAccessUnit() {
 		super(Token.class, Token.ENTITY_NAME);
 	}
+	private static final Logger LOG = LoggerFactory.getLogger(GuildSettingAccessUnit.class);
 
 	public List<Token> forKey(TokenKey key) {
 		return get(Collections.singletonMap("key", key));
 	}
+
+	public void update(Token token) {
+		try {
+			List<Token> foundTokens = forKey(token.getKey());
+
+			foundTokens.get(0).setToken(token.getToken());
+			retryOnFail(() -> dao.update(TokenPO.class, foundTokens.get(0), foundTokens.get(0).getId()));
+
+		} catch(FailedAfterRetriesException e) {
+			LOG.error("Error while trying to add entity of type GuildSettingsPO");
+			LOG.debug("Stacktrace:", e);
+		}
+	}
```

Comment:
This should not be needed anymore. The update function is provided by this classes parent (StorageAccessUnit<Token>)